clustered/client/src/main/java/org/ehcache/clustered/client/internal/config/xml/ClusteringServiceConfigurationParser.java

```diff
@@ -93,6 +119,16 @@ public URI getNamespace() {
         fragment.getTagName(), (fragment.getParentNode() == null ? "null" : fragment.getParentNode().getLocalName())));
   }
 
+  @Override
+  public Class<ClusteredStore.Provider> handleServiceConfigurationType() {
```

Comment:
This single class is so huge since it's still handling the parsing and translation of both `ClusteredStoreConfiguration` and `ClusteringServiceConfiguration`. Splitting this class to handle both separately would get us some smaller simpler classes. I know that it wasn't your decision to make it implement both `CacheManagerServiceConfigurationParser` and `CacheServiceConfigurationParser`. But looking at the current size it feels like it's time to split.

Reply:
I can make the changes. Please let me know if you want the refactoring to be done as part of same PR

Reply:
It can be a follow-up PR if that's too much work. But I feel that it isn't.

Reply:
Ok I will split this into two classes. one implementing CacheManagerServiceConfigurationParser and other CacheServiceConfigurationParser.